src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -149,9 +144,13 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             // WSLENV is a colon-delimited list of environment variables (+flags) that should appear inside WSL
             // https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/
 
-            auto wslEnv = environment[L"WSLENV"];
-            wslEnv = L"WT_SESSION:" + wslEnv; // prepend WT_SESSION to make sure it's visible inside WSL.
-            environment.insert_or_assign(L"WSLENV", wslEnv);
+            const auto existingWslEnv = environment.as_map()[L"WSLENV"];
+            std::wstring wslEnv{ L"WT_SESSION:WT_PROFILE_ID" };
+            if (!existingWslEnv.empty())
+            {
+                wslEnv += L":" + existingWslEnv; // prepend WT_SESSION and WT_PROFILE_ID to make sure they're visible inside WSL.
```

Comment:
it's harmless to append an empty `existingWslEnv`, FWIW, if you are seeking to simplify